flake.nix

```diff
@@ -0,0 +1,55 @@
+{
+  description = "Simon’s Improved Layout Engine";
+
+  # To make user overrides of the nixpkgs flake not take effect
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+
+  # https://nixos.wiki/wiki/Flakes#Using_flakes_project_from_a_legacy_Nix
+  inputs.flake-compat = {
+    url = "github:edolstra/flake-compat";
+    flake = false;
+  };
+
+  outputs = { self
+    , nixpkgs
+    , flake-utils
+    , flake-compat
+  }:
+  flake-utils.lib.eachDefaultSystem (system:
+    let
+      pkgs = import nixpkgs {
+        inherit system;
+      };
+      # Use the expression from Nixpkgs instead of rewriting it here.
+      sile = pkgs.sile.overrideAttrs(oldAttr: {
+        version = "${(pkgs.lib.importJSON ./package.json).version}-flake";
+        # TODO: Building from source doesn't work yet as the fonts are not
+        # available during build and can't be downloaded in the sandbox.
```

Comment:
Could we add the fonts into `buildInputs` (maybe as in #1223's FONTCONFIG_FILE)? (And patch out `include Makefile-fonts` + `LOCALTESTFONTS` if necessary, again maybe as in #1223?)

Honestly, for me personally, the whole point of needing a flake.nix is to be able to build locally to make development on #220 easier...

Reply:
> Honestly, for me personally, the whole point of needing a flake.nix is to be able to build locally to make development on #220 easier...

Does sile need certain fonts available during sile's build in order to test math support? Or could they be available only when sile is building documents?

Reply:
Not to build no. If you want to run the regression test suite then it needs not only a MATH font but a whole army of other fonts besides.

However the default math font is *Libertinus Math*, so if people try to test any math stuff they will either need to use a font they have or get that. I don't know if we should start baking that is as a requirement, but since it is used in a package not in core I think it's probably okay that we just make the user supply a font rather than making everybody install one as a dependency.

Reply:
We could suggest a `FONTCONFIG_FILE` in `.envrc` or in `devShell`, but that could be part of a future PR. The only issue is that Nix users don't like fonts to be downloaded during build, so hopefully we'll be able to rely upon fonts from Nixpkgs.

Reply:
Understood, and yes I agree.

By default the build system shouldn't muck about with downloading any fonts at all provided the manual and tests are both disabled at configure time. The only other thing that is *optionally* online during the build is the vendored Luarocks option, but that's irrelevant to Nix since it uses system supplied dependencies for those already.